src/grt/src/GlobalRouter.cpp

```diff
@@ -1363,6 +1378,55 @@ void GlobalRouter::setMacroExtension(int macroExtension)
   _macroExtension = macroExtension;
 }
 
+void GlobalRouter::setCapacitiesPerturbationPercentage(float percentage)
+{
+  caps_perturbation_percentage_ = percentage;
+}
+
+void GlobalRouter::perturbCapacities()
+{
+  int xGrids = _grid->getXGrids();
+  int yGrids = _grid->getYGrids();
+
+  int num_2d_grids = xGrids*yGrids;
+  int num_perturbations = caps_perturbation_percentage_*num_2d_grids;
+
+  for (int layer = 1; layer <= _maxRoutingLayer; layer++) {
+    std::mt19937 g;
+    g.seed(seed_);
```

Comment:
I would put this outside the loop so that you perturb the layers without correlation.